dhcpv4/nclient4/client.go

```diff
@@ -162,6 +162,18 @@ type Client struct {
 	// TransactionID. receiveLoop uses this map to determine which channel
 	// to send a new DHCP message to.
 	pending map[dhcpv4.TransactionID]*pendingCh
+
+	//clientIdOptions is a list of DHCPv4 option code that DHCP server used to
+	//identify client other than the HWAddress,
+	//like client-id, option82/remote-id..etc
+	clientIDOptions dhcpv4.OptionCodeList
```

Comment:
The only option that would make sense to have here is the client-id one (61). I don't think it makes sense to have the relay agent info here (82). Also, the proper parsing of option 61 is not implemented in the library. Should it be implemented before using it here?

Reply:
using option82 (circuit-id/remote-id) to identify client is a common practice by broadband service provider (as described RFC3046 section 1); the idea of clientIDOptions is to have a flexible way that could work with many different DHCP server requirements.

I could also take a look on option61 parsing

Reply:
Sure, it can be used by the server but it shouldn't be inserted by the client.

As per the RFC:

> The new option is called the Relay Agent Information option and is inserted by the DHCP relay agent

So this is not a normal client behaviour.

Reply:
yes, a DHCP client on end user device will not insert option82; however if someone want to use this library to write a testing client to test DHCP server that requires option82 without any access node, then it is useful in such case (actually that's my use case)
plus clientIDOptions is not hard-coded to option82 or any specific option; it could be empty  if it is not needed;

Reply:
This feels like you are tuning the client to your particular needs more than implementing standard and RFC compliant things.

Can you not use simply use a modifier?

Reply:
the problem of using a modifier is when client need to send DHCP release, it needs to use same ID option as used during DORA, otherwise server won't be able to identify the lease to release; and in case when persistent lease storage is needed, e.g. save lease to a file so that it could be read later on to create the client and send the release, the ID option need to be saved as part of lease; 
maybe it is not a good idea of using option82/remote-id in my example, but clientIDOptions is not really only for option82;

Reply:
How about renaming this to `copyOptions`? That is what this is for, isn't it?And it is not specific to ID options.